### PR TITLE
Upgrade to Realm Core v10.6.0

### PR DESCRIPTION
**Pull request by kneth**
## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* ~[ ] 🚦 Tests~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+ 
 
 ### Fixed
 * Using node version 15, the error `sh: cmake-js: command not found` will prevent installation. ([#3670](https://github.com/realm/realm-js/issues/3670), since v10.3.0-rc.1)
 * On React Native, calling an aggregate function would fail with error `Not implemented`. ([#3674](https://github.com/realm/realm-js/issues/3674), since v10.2.0)
+* Fixed name aliasing (`mapTo` in schemas) not working in sort/distinct clauses of the query language. ([realm/realm-core#4550](https://github.com/realm/realm-core/issues/4550), never worked)
+* Potential/unconfirmed fix for crashes associated with failure to memory map (low on memory, low on virtual address space). ([realm/realm-core#4514](https://github.com/realm/realm-core/issues/4514))
+* Fixed collection notification reporting for modifications. This could be observed by receiving the wrong indices of modifications on sorted or distinct results, or notification blocks sometimes not being called when only modifications have occured. ([realm/realm-core#4573](https://github.com/realm/realm-core/pull/4573), since v6.0.0)
```
Comment (bmunkholm) [inline CHANGELOG.md:12]: Does 6.0.0 refer to core or JS version? It should be JS version.

**Review by bmunkholm — APPROVED** [review]
LGTM after checking/updating JS version used with core 6.0.0

File: CHANGELOG.md (hunk 1 above)
Comment (kneth) [reply CHANGELOG.md:12]: Yes, v6.0.0 is a special version number as JS v6.0.0 use Core v6.0.0 :-)

**Review by steffenagger — APPROVED** [review]
LGTM

**Pull request merged**